hppc/src/main/templates/com/carrotsearch/hppc/KTypeArrayList.java

```diff
@@ -278,19 +301,18 @@ public int removeAll(KType e1) {
     int to = 0;
     for (int from = 0; from < elementsCount; from++) {
       if (Intrinsics.equals(this, e1, buffer[from])) {
-        buffer[from] = Intrinsics.empty();
         continue;
       }
-
       if (to != from) {
         buffer[to] = buffer[from];
-        buffer[from] = Intrinsics.empty();
       }
       to++;
     }
-
     final int deleted = elementsCount - to;
     this.elementsCount = to;
+    /* #if ($TemplateOptions.KTypeGeneric) */
+    Arrays.fill(buffer, elementsCount, elementsCount + deleted, Intrinsics.empty());
```

Comment:
Do you think it's going to be faster this way? Those null assignments should be within the same cache line, so super-fast already? And previously it'd zero the removed slot, now it doesn't. Don't know if it makes a difference in practice.

Reply:
Since you asked the question, I did a benchmark running removeAll on lists of size 10, 100, 1_000, 10_000, and with four categories of number of duplicate elements to make the removal longer.
- Surprisingly (to me) I see not perf difference for primitive IntArrayList with and without zeroing removed slots.
- For ObjectArrayList, the new version with Arrays.fill is 20% faster.

So I would tend to keep this new code, unless you see an interest in zeroing removed slots for primitives. BTW the KTypeSlack implementation of pop was already not zeroing removed slots, and that suggested me to do the same for list.

Reply:
Thank you for running the benchmark. LGTM.